src/Compilers/CSharp/Test/Syntax/Syntax/SyntaxFactoryTests.cs

```diff
@@ -456,5 +456,81 @@ public void GetDiagnosticsFromNullToken()
             Assert.Equal(Location.None, token.GetLocation());
             token.GetDiagnostics().Verify();
         }
+
+        [Fact]
+        [WorkItem(21231, "https://github.com/dotnet/roslyn/issues/21231")]
+        public void TestSpacingOnNullableIntType()
+        {
+            var syntaxNode =
+                SyntaxFactory.CompilationUnit()
+                .WithMembers(
+                    SyntaxFactory.SingletonList<MemberDeclarationSyntax>(
+                        SyntaxFactory.ClassDeclaration("C")
+                        .WithMembers(
+                            SyntaxFactory.SingletonList<MemberDeclarationSyntax>(
+                                SyntaxFactory.PropertyDeclaration(
+                                    SyntaxFactory.NullableType(
+                                        SyntaxFactory.PredefinedType(
+                                            SyntaxFactory.Token(SyntaxKind.IntKeyword))),
+                                    SyntaxFactory.Identifier("P"))
+                                    .WithAccessorList(
+                                        SyntaxFactory.AccessorList())))))
+                .NormalizeWhitespace();
+
+            // no space between int and ?
+            Assert.Equal("class C\r\n{\r\n    int? P\r\n    {\r\n    }\r\n}", syntaxNode.ToFullString());
```

Comment:
>\r\n [](start = 33, length = 4)

I think the test will be more robust if it would use platform specific line breaks, Consider using verbatim string literal instead.

Reply:
When I used a verbatim string in an earlier commit, the tests failed on Linux due to line breaks.
I'll try to see if we have another assert helper for dealing with that.

Reply:
From discussion with Andy, the issue with line terminators on Linux is known and not trivial to fix (we need to decide what the behavior should be when modifying syntax in a predominantly Windows-terminated file on Linux). 
I added more details to https://github.com/dotnet/roslyn/issues/8537 as follow-up issue.